### ISSUE TITLE
Should List<T> keep its initializer that takes a sequence?

### ISSUE THREAD

**Issue by bdash** [realm/realm-cocoa#3691]
Due to `List<T>`'s conformance to `RangeReplaceableCollectionType` it picks up an initializer that takes a sequence:

``` swift
func init<S : SequenceType where S.Generator.Element == Generator.Element>(_: S)
```

This allows constructs of questionable value such as `List<Person>([personA, personB])`, and may set users up for failure by encouraging them to attempt to assign to `List<T>` properties.

We could:
1. Remove conformance to `RangeReplaceableCollectionType` while providing implementations of the remaining methods it declares.
2. Override the initializer to abort with a useful error.
3. Propose removing the initializer from the protocol on swift-evolution since it's not clear how it fits the described purpose of the protocol.
4. Do nothing.

Thoughts?

**Comment by austinzheng** [comment]
Is there any reason to conform `List` to `RangeReplaceableCollectionType` besides getting whatever methods it provides for free? I'm not sure if there are any good use cases for clients being allowed to use `Lists` interchangeably with other range replaceable collection types, especially because the 'range replaceable' behavior in question requires the API calls to be wrapped in a write transaction.

**Comment by jpsim** [comment]
I'd propose doing options 2 _and_ 3. Thinking about this more since our Slack conversation this morning, I think it's useful to keep our conformance to the protocol, here's why:

> Is there any reason to conform `List` to `RangeReplaceableCollectionType` besides getting whatever methods it provides for free?

Yes, generic programming. Say you had an `RangeReplaceableCollectionType.inPlaceSort()` function, you could then use it on `List`. You'd still need to wrap calls to that in a write transaction, but that's the same as any other mutating operation today.

**Comment by austinzheng** [comment]
Fixed in #5276.
